src/containers/orgs/OrgsContainer.js

```diff
@@ -132,27 +156,80 @@ class OrgsContainer extends Component<Props> {
 
   componentDidUpdate(props :Props) {
 
-    const { actions, requestStates } = this.props;
-    if (requestStates[GET_ORGS_AND_PERMISSIONS] === RequestStates.SUCCESS
-        && props.requestStates[GET_ORGS_AND_PERMISSIONS] === RequestStates.PENDING) {
+    const { actions, newlyCreatedOrgId, requestStates } = this.props;
+
+    if (props.requestStates[GET_ORGS_AND_PERMISSIONS] === RequestStates.PENDING
+        && requestStates[GET_ORGS_AND_PERMISSIONS] === RequestStates.SUCCESS) {
       actions.resetRequestState(OrgsActions.GET_ORGS_AND_PERMISSIONS);
     }
+
+    if (props.requestStates[CREATE_ORGANIZATION] === RequestStates.PENDING
+        && requestStates[CREATE_ORGANIZATION] === RequestStates.SUCCESS) {
+      this.goToOrg(newlyCreatedOrgId);
+    }
   }
 
   componentWillUnmount() {
 
     const { actions } = this.props;
+    actions.resetRequestState(CREATE_ORGANIZATION);
     actions.resetRequestState(OrgsActions.GET_ORGS_AND_PERMISSIONS);
   }
 
-  goToOrg = (org :Map) => {
+  goToOrg = (orgId :?UUID) => {
+
     const { actions } = this.props;
-    const orgId :UUID = org.get('id');
-    actions.goToRoute(Routes.ORG.replace(Routes.ID_PARAM, orgId));
+
+    if (isValidUUID(orgId) && orgId) {
+      actions.goToRoute(Routes.ORG.replace(Routes.ID_PARAM, orgId));
+    }
+    else {
+      LOG.error('organization id is not a valid UUID', orgId);
+    }
+  }
+
+  handleOnChangeNewOrgTitle = (event :SyntheticInputEvent<HTMLInputElement>) => {
+
+    this.setState({ newOrgTitle: event.target.value || '' });
+  }
+
+  handleOnClickCreateOrg = () => {
+
+    const { actions } = this.props;
+    const { newOrgTitle } = this.state;
+
+    if (isNonEmptyString(newOrgTitle)) {
+
+      const principal = (new PrincipalBuilder())
+        .setType(PrincipalTypes.ORGANIZATION)
+        .setId(newOrgTitle.replace(/\W/g, ''))
+        .build();
+
+      const org = (new OrganizationBuilder())
+        .setTitle(newOrgTitle)
+        .setPrincipal(principal)
+        .build();
+
+      actions.createOrganization(org);
+    }
+  }
+
+  closeNewOrgModal = () => {
+
+    const { actions } = this.props;
+
+    this.setState({ isVisibleNewOrgModal: false, newOrgTitle: '' });
+    actions.resetRequestState(CREATE_ORGANIZATION);
+  }
+
+  openNewOrgModal = () => {
+
+    this.setState({ isVisibleNewOrgModal: true, newOrgTitle: '' });
   }
 
   renderOrgCard = (org :Map) => {
 
+    const orgId :UUID = org.get('id');
     let description :string = org.get('description', '');
 
     // TODO: refactor as a utility function
```

Comment:
Looks like you're trying to show a snippet of text truncated by ellipses at the end. Consider `--webkit-line-clamp` for multi line blocks that end in ellipses.

https://developer.mozilla.org/en-US/docs/Web/CSS/-webkit-line-clamp

Reply:
😍

Reply:
Done!